src change - for gh-pages

emilyville opened the pull request:
- In order to generate github pages, had to move everything out of the src/ and into top level.
- also added excludes for the _site/
- updated baseurl

Commit:
use default top-level as the source directory

Commit:
exclude source files from webserver, set baseurl

Commit:
remove _config.yml from excludes list

Commit:
add trailing slash to site url

Commit:
update excludes

webinista commented:
Looks like one of the default posts is missing a valid date in the YAML front matter, so Jekyll can't build the site. There are also some merge conflicts.

@awong-dev I tried merging, but this appears to delete everything. Do you have time to take a look?

awong-dev commented:
Gha... @webinista I completely missed this.  If you assign me a PR, can you possibly ping me on slack?  I get so many github notifications, I have a tendency to miss these.  If you haven't seen anything from me in like an hour, you can be pretty sure it's been lost. :(

I'll try to get better at watching my assigned queue, but always feel free to pester if I'm not responding.

Commit:
exclude vendor

Commit:
put site in /roadrunner subdirectory

Commit:
fix footer link for htmlproof, and add some new rake targets

emilyville commented:
ok @webinista or @awong-dev all green again!
- It was pulling in a default post file from the gem - so I added an vendor exclude to fix that
- Then, the html checker was expecting a different structure
- then, the links with a "/" were throwing it off so changed to "#"
- added rake targets for better shortcuts

Commit:
moar better footer link

emilyville commented:
Made the footer link correct instead of just #. :)

Commit:
url in config file should not include baseurl

emilyville commented:
Removed the "roadrunner" baseurl from the url to fix the external link html proofer tests.

awong-dev commented on Rakefile:
```diff
@@ -5,6 +5,16 @@ task :build do
+
+desc "Cleanup the website"
+task :clean do
+  sh "bundle exec jekyll clean"
```
? I don't think this is a real jekyll command?

emilyville replied:
it basically just deletes the contents of the destination directory

http://www.rubydoc.info/github/mojombo/jekyll/Jekyll/Commands/Clean

awong-dev replied:
Interesting...my version of jekyll doesn't have it. I must be out of date.

awong-dev commented on _config.yml:
```diff
@@ -3,18 +3,23 @@ title: Roadrunner
 markdown: kramdown
 
-destination: ./_site
+destination: ./_site/roadrunner
```
Is this correct?

awong-dev commented:
@emilyville back at ya.

awong-dev commented:
LGTM w/o caveat.

I'm still uncertain about `destination: ./_site/roadrunner` but this PR has been outstanding too long and it's less harmful to have that possible mistake checked in than leaving it open.
